models/fact.js

```diff
@@ -271,6 +272,113 @@ function getFact({ number, type, options = {} }) {
     });
   }
 }
+/**
+ * This function mutates the given empty object to add keys for each type
+ * This is a helper function for getAllFacts and getTypeFacts
+ *
+ * start {}
+ * ends {type: [], type:[]}
+ *
+ * @param types: string[],
+ * @param object: empty object
+ * @param dateNum: number converted to conform to date number
+ * @param num: current number
+ *
+ *
+ */
+function getFactTexts(types, object, dateNum, num) {
+  for (let type of types) {
+    if (type === "date" && data[type][dateNum]) {
+      let prefix = utils.getStandalonePrefix(dateNum, type);
+      object[type] = data[type][dateNum].map(({ text }) => `${prefix} ${text}`);
+    } else if (data[type][num]) {
+      object[type] = data[type][num].map(({ text }) => text);
+    } else {
+      object[type] = [getDefaultMsg({ number: num, type })];
+    }
+  }
+}
+
+/**
+ * getAllFacts() is used as a graphQL endpoint in lieu of querying a database
+ *
+ * @param num: is an int
+ * @returns {
+ *            number: num
+ *            year: string[],
+ *            trivia: string[],
+ *            math: string[],
+ *            date: string[]
+ *          }
+ * The return must conform to the GraphQL NumberType expected fields, which are currently {number, year, trivia, math, date}
+ */
+function getAllFacts(num) {
+  let res = {};
+  let types = ["year", "trivia", "math", "date"];
+  let dateNum = utils.dateToDayOfYear(new Date(2004, 0, num));
+
+  getFactTexts(types, res, dateNum, num);
+
+  res.number = num;
+  return res;
+}
+/**
+ *  getTypeFacts() is used as a graphQL endpoint in leiru of querying a database
+ *
+ * @param types: string[], accepts "trivia", "math", "year", "date" as accepted strings
+ * @param numbers: int[]
+ *
+ * @returns
+ *  [
+ *    {
+ *      num: {
+ *         types:
+ *           {
+ *              type: [],
+ *              type: []
+ *           }
+ *      },
+ *      num: {
```

Comment:
The detail in the docstrings makes the function easier to understand but it was a little confusing because num and type are dynamic but but types is not.
Lets wrap the num and type with computed properties notation to help signal it is different than the types key.
```
{
 *      [num]: {
 *         types:
 *           {
 *              [type]: [],
 *              [type]: []
 *           }
 *      }
 ```

Reply:
Got it! Updated the doc string to include [] around dynamic keys